### PR TITLE
[RPC] server should report client errors promptly

### PR DESCRIPTION
**Pull request by rgrinberg**
Currently, it waits until the client disconnected before reporting errors

@jeremiedimino I'm not sure how to report such errors correctly. They are not
really the fault of the developer or the user, but more like errors caused by
invalid clients. It's useful to see them on the server side however.

### CODE REVIEW

File: src/dune_rpc_server/dune_rpc_server.ml [hunk 1]
```diff
@@ -419,15 +422,24 @@ struct
         in
         let id = session#id in
         Event.emit (Session Start) stats id;
-        let+ res = Fiber.collect_errors (fun () -> session#start) in
+        let+ res =
+          Fiber.collect_errors (fun () ->
```
Comment (jeremiedimino) [inline src/dune_rpc_server/dune_rpc_server.ml:426]: You can use a `map_reduce_errors` here, as in `Scheduler.Run.poll`. That should simplify this code a bit.

**Comment by jeremiedimino** [comment]
it still seems useful to show them to the user, so that they are aware there is an issue and can report it.

File: src/dune_rpc_server/dune_rpc_server.ml (hunk 1 above)
Comment (rgrinberg) [reply src/dune_rpc_server/dune_rpc_server.ml]: Thanks for the tip.

**Comment by rgrinberg** [comment]
> it still seems useful to show them to the user, so that they are aware there is an issue and can report it

That's what I suspected. The PR now reports the errors properly.

File: src/dune_rpc_server/dune_rpc_server.ml [hunk 2]
```diff
@@ -419,15 +419,29 @@ struct
...
+        in
         Event.emit (Session Stop) stats id;
         match res with
         | Ok () -> ()
-        | Error exns ->
-          Dune_util.Log.info
-            [ Pp.text "fatal error when serving rpc client"
-            ; Pp.concat_map ~sep:Pp.newline exns ~f:Exn_with_backtrace.pp
-            ])
+        | Error _exn ->
```
Comment (jeremiedimino) [inline src/dune_rpc_server/dune_rpc_server.ml:442]: Might as well be explicit here:

```suggestion
        | Error () ->
```

**Review by jeremiedimino — APPROVED** [review]
Looks correct to me

**Pull request merged**